vscode/oso/client/src/index.ts

```diff
@@ -211,17 +228,22 @@ function updateClients(context: ExtensionContext) {
   };
 }
 
-const flushTelemetryEvents = () =>
-  flushQueue().catch(e =>
-    outputChannel.appendLine(`Caught error while sending telemetry: ${e}`)
-  );
+let persistState: (state: TelemetryCounters) => Promise<void> = async () => {}; // eslint-disable-line @typescript-eslint/no-empty-function
 
 export async function activate(context: ExtensionContext): Promise<void> {
+  const state = context.globalState.get<TelemetryCounters>(TELEMETRY_STATE_KEY);
+  persistState = async (state: TelemetryCounters) =>
+    context.globalState.update(TELEMETRY_STATE_KEY, state);
+  seedState(state);
```

Comment:
Capturing `context.globalState` in this closure since we won't have access to it in `deactivate()`, where we want to persist the updated state.